app/models/user.rb

```diff
@@ -908,7 +908,7 @@ def username_format_validator
   def email_confirmed?
     email_tokens.where(email: email, confirmed: true).present? ||
     email_tokens.empty? ||
-    single_sign_on_record&.external_email == email
+    single_sign_on_record&.external_email&.downcase == email
```

Comment:
is `email` already downcased here? I wonder if it should be 

`single_sign_on_record&.external_email&.downcase == email.downcase`

Reply:
Yes, `email` is downcased there. My understanding is that user emails are always downcased before being stored to the database. They are downcased by the `strip_downcase_email` `before_validation` filter.